### PR TITLE
Add header `titlePadding` and `labelPadding`

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/header.ts [hunk 1]
```diff
@@ -82,6 +85,20 @@ export interface HeaderConfig {
    */
   titleLimit?: number;
 
+  /**
+   * The orthogonal offset in pixels by which to displace the title from its position along the edge of the chart.
+   *
+   * __Default value:__ `10`, indicating no limit
+   */
+  titleOffset?: number;
```
Comment (kanitw) [inline src/header.ts]: A similar thing is called titlePadding in Axis.

File: src/header.ts [hunk 2]
```diff
@@ -116,6 +133,20 @@ export interface HeaderConfig {
    * __Default value:__ `0`, indicating no limit
    */
   labelLimit?: number;
+
+  /**
+   * The orthogonal offset in pixels by which to displace the title from its position along the edge of the chart.
+   *
+   * __Default value:__ `10`, indicating no limit
+   */
+  labelOffset?: number;
```
Comment (kanitw) [inline src/header.ts]: A similar thing is called labelPadding in Axis.

File: src/header.ts [hunk 3]
```diff
@@ -82,6 +85,20 @@ export interface HeaderConfig {
...
+   * __Default value:__ `10`, indicating no limit
+   */
+  titleOffset?: number;
+
+  /**
+   * Title orientation (`"top"`, `"bottom"`, `"left"`, or `"right"`).
+   *
+   * __Default value:__ `"left"` for rows and `"top"` for columns.
+   */
+  titleOrient?: TitleOrient;
```
Comment (kanitw) [inline src/header.ts]: Does mapping title orient to orient of underlying title work?

Don't you have to do much more (correctly applying to header/footer, etc.)?

File: src/header.ts [hunk 4]
```diff
@@ -116,6 +133,20 @@ export interface HeaderConfig {
...
+   * __Default value:__ `10`, indicating no limit
+   */
+  labelOffset?: number;
+
+  /**
+   * Label orientation (`"top"`, `"bottom"`, `"left"`, or `"right"`).
+   *
+   * __Default value:__ `"left"` for rows and `"top"` for columns.
+   */
+  labelOrient?: TitleOrient;
```
Comment (kanitw) [inline src/header.ts]: Does mapping label orient to orient of underlying title work?

I thin you have to do much more (correctly applying to header/footer, etc.)?

File: src/header.ts (hunk 3 above)
Comment (domoritz) [reply src/header.ts]: Done

File: src/header.ts (hunk 1 above)
Comment (domoritz) [reply src/header.ts]: Done

File: src/header.ts [hunk 5]
```diff
@@ -82,6 +84,13 @@ export interface HeaderConfig {
    */
   titleLimit?: number;
 
+  /**
+   * The orthogonal distance in pixels by which to displace the title from its position along the edge of the chart.
+   *
+   * __Default value:__ `10`, indicating no limit
```
Comment (kanitw) [inline src/header.ts]: Why is 10 "no limit"

File: src/header.ts [hunk 6]
```diff
@@ -116,6 +125,13 @@ export interface HeaderConfig {
    * __Default value:__ `0`, indicating no limit
    */
   labelLimit?: number;
+
+  /**
+   * The orthogonal distance in pixels by which to displace the title from its position along the edge of the chart.
+   *
+   * __Default value:__ `10`, indicating no limit
```
Comment (kanitw) [inline src/header.ts]: Why is 10 "no limit"

**Pull request merged**